packages/dd-trace/src/appsec/blocking.js

```diff
@@ -100,29 +100,43 @@ function getBlockingData (req, specificType, actionParameters) {
 }
 
 function block (req, res, rootSpan, abortController, actionParameters = defaultBlockingActionParameters) {
-  if (res.headersSent) {
-    log.warn('[ASM] Cannot send blocking response when headers have already been sent')
-    return
-  }
+  let blocked
+  try {
+    if (res.headersSent) {
+      log.warn('[ASM] Cannot send blocking response when headers have already been sent')
 
-  const { body, headers, statusCode } = getBlockingData(req, null, actionParameters)
+      throw new Error('Headers have already been sent')
+    }
 
-  rootSpan.addTags({
-    'appsec.blocked': 'true'
-  })
+    const { body, headers, statusCode } = getBlockingData(req, null, actionParameters)
 
-  for (const headerName of res.getHeaderNames()) {
-    res.removeHeader(headerName)
-  }
+    for (const headerName of res.getHeaderNames()) {
+      res.removeHeader(headerName)
+    }
 
-  res.writeHead(statusCode, headers)
+    res.writeHead(statusCode, headers)
 
-  // this is needed to call the original end method, since express-session replaces it
-  res.constructor.prototype.end.call(res, body)
+    // this is needed to call the original end method, since express-session replaces it
+    res.constructor.prototype.end.call(res, body)
 
-  responseBlockedSet.add(res)
+    rootSpan.addTags({
+      'appsec.blocked': 'true'
+    })
+
+    responseBlockedSet.add(res)
+    abortController?.abort()
+
+    blocked = true
+  } catch (err) {
+    rootSpan?.addTags({
+      '_dd.appsec.block.failed': 1
+    })
+    log.error('[ASM] Blocking error', err)
+
+    blocked = false
+  }
 
-  abortController?.abort()
+  return blocked
```

Comment:
no need for a variable, just add return true in the try, and return false in the catch